api/v1alpha1/k8ssandracluster_types.go

```diff
@@ -76,42 +76,77 @@ type K8ssandraClusterList struct {
 	Items           []K8ssandraCluster `json:"items"`
 }
 
-type Cassandra struct {
+type CassandraClusterTemplate struct {
+	// Cluster is the name of the cluster. This corresponds to cluster_name in
+	// cassandra.yaml.
+	// +kubebuilder:validation:MinLength=2
 	Cluster string `json:"cluster,omitempty"`
 
-	Datacenters []CassandraDatacenterTemplateSpec `json:"datacenters,omitempty"`
+	// ServerImage is the image for the cassandra container. Note that this should be a
+	// management-api image. If left empty the operator will choose a default image based
+	// on ServerVersion.
+	ServerImage string `json:"serverImage,omitempty"`
```

Comment:
Should probably be +optional (cass-operator's choice of image should be the newest one released for that ServerVersion)